### PR TITLE
Themeing

### PR DESCRIPTION
**Pull request by dimitritsampiras**
1. Wraps apollo client provider in single component to remove all the config away from App.tsx
2. Adds Heading, SubHeading components 
3. Fixes major issue where jwt stores user object. However, jwt doesn't change when user updates in db so data is out of sync. JWT only stores the userId now so the 'me' query fetches the user from the database. 
4. tsconfig rules 
5. adds global screen component to be used in replace of "SafeAreaView"
6. misc config stuff

### CODE REVIEW

**Comment by bentvelj** [comment]
In the future can we try to keep PRs focused on a single issue, this is like 6 PRs.


guh..
image
image

File: src/athena/src/components/containers/TabBar.tsx [hunk 1]
```diff
@@ -1,85 +1,79 @@
...
-  BottomTabBarProps,
-  BottomTabNavigationOptions
-} from '@react-navigation/bottom-tabs';
-import React from 'react';
-import { Pressable, StyleSheet, Text, TouchableOpacity, View } from 'react-native';
-import * as Haptics from 'expo-haptics';
-import { Cog6ToothIcon, HomeIcon } from 'react-native-heroicons/solid';
-import theme from '../../theme';
-import Svg, { Path } from 'react-native-svg';
+// import { BottomTabBarProps, BottomTabNavigationOptions } from '@react-navigation/bottom-tabs';
```
Comment (bentvelj) [inline src/athena/src/components/containers/TabBar.tsx:1]: Why we commenting all this out? Should be deleted, old code should be accessed through git

**Comment by dimitritsampiras** [comment]
Future prs will be smaller for sure.

File: src/athena/src/components/screens/auth/SignUp.tsx [hunk 2]
```diff
@@ -12,9 +12,10 @@ import {
 import { SafeAreaView } from 'react-native-safe-area-context';
 import { AuthParamList } from '../../navigation';
 import {
-  useFindEmailMutation,
-  useFindUsernameMutation,
+  useEmailExistsLazyQuery,
   useSignUpMutation,
+  useUserExistsLazyQuery,
+  useUserExistsQuery,
```
Comment (bentvelj) [inline src/athena/src/components/screens/auth/SignUp.tsx:18]: For my own understanding, what is a LazyQuery?

File: src/athena/src/graphql/fragments/User.graphql [hunk 3]
```diff
@@ -0,0 +1,6 @@
+fragment User on User {
+  id
+  name
+  username
+  email
+}
```
Comment (bentvelj) [inline src/athena/src/graphql/fragments/User.graphql:6]: For my own understanding, what's a fragment?

File: src/hermes/src/server/schema/User/helpers/authentication.ts [hunk 4]
```diff
@@ -55,15 +59,23 @@ export const loginUser = async (prisma: PrismaClient, input: LoginInput) => {
 
   // user does not exist guard clause
   if (!user) return null;
+  console.log(user);
+
```
Comment (bentvelj) [inline src/hermes/src/server/schema/User/helpers/authentication.ts:63]: We should make these more descriptive, it's annoying to boot up the app and have the console flooded with a bunch of `undefined`, `null` or random objects without context.

Furthermore, we should use proper logging, but console.log will work for now, and I'll create an issue for logging

File: src/hermes/src/server/schema/User/user.mutation.ts [hunk 5]
```diff
@@ -14,32 +13,16 @@ export const UserMutation = extendType({
       type: nullable('String'),
       args: { input: 'LoginInput' },
       resolve: async (_root, { input }, { prisma }) => {
-        const token = await loginUser(prisma, input);
-        return token;
+        console.log('hit');
```
Comment (bentvelj) [inline src/hermes/src/server/schema/User/user.mutation.ts:16]: Descriptive

File: src/hermes/src/server/schema/User/helpers/authentication.ts [hunk 6]
```diff
@@ -55,15 +59,23 @@ export const loginUser = async (prisma: PrismaClient, input: LoginInput) => {
...
-
-  return token;
+  try {
+    const token = jwt.sign({ userId: user.id }, config.jwtSecret, {
+      expiresIn: '3d',
+    });
+
+    return token;
+  } catch (error) {
+    console.log(error);
```
Comment (bentvelj) [inline src/hermes/src/server/schema/User/helpers/authentication.ts:76]: Descriptive

File: src/athena/src/components/screens/auth/SignUp.tsx (hunk 2 above)
Comment (dimitritsampiras) [reply src/athena/src/components/screens/auth/SignUp.tsx:18]: Regular queries fire as soon as the component renders. Lazy queries are for when you want to call it manually like on a button click.

File: src/athena/src/graphql/fragments/User.graphql (hunk 3 above)
Comment (dimitritsampiras) [reply src/athena/src/graphql/fragments/User.graphql:6]: you might make a few queries like this at some point:

```
query {
    allPrograms {
        name
        trainingLevel
        ratings
    }
}

query {
    programsOfUser {
        name
        trainingLevel
        ratings
    }
}

query {
    programFromID {
        name
        trainingLevel
        ratings
    }
}
```

You can use a fragment to standardize the return types of each query so you're not repeating the same stuff over and over.

**Pull request merged**